docs/partners_pytorch_ci_runners.md

```diff
@@ -0,0 +1,55 @@
+# Partners PyTorch CI Runners
+
+
+This document tries to facilitate and guide partners who want to contribute with PyTorch CI providing runners. This guide is mostly intended for a static fleet of runners. In case the goal is either to use ARC or the `terraform-aws-github-runner` module provided in this repository this does not apply.
+
+
+## Overall view of the process
+
+
+### Reasoning
+
+
+Due to the past a big number of stale runners, we enforce that any offline CI runner is automatically deleted from the list of available CI runners for PyTorch. This should not introduce any work for dynamically scaled instances, quite the opposite, as it allows the developer to not worry about cleanup and stale instances. But it can be a bit extra work for providers of a static fleet of instances.
+
+
+Another advantage of having this is the fact that it enables PyTorch Foundation to quickly isolate and remove runners that are problematic or suffered a security incident, even if they are dynamically scaled (so deleting one-by-one is not an option).
+
+
+### Overview
+
+
+To register runners, the partner should create a github app visible to PyTorch org in GH, requesting the relevant read and write access to organization self hosted runners. Note that this is the exact same steps required to setup ARC or `terraform-aws-github-runner`. But for this case there is no need to subscribe to any events.
+
+
+As a next step, the partner should reach out to one of the PyTorch org admins and request the installation of the app, making sure its scope is limited to `pytorch/pytorch` repository.
```

Comment:
Might be good to suggest to post an issue on `pytorch/test-infra` so that we don't have to reach out to specific people